superset-frontend/packages/superset-ui-core/src/query/types/Column.ts

```diff
@@ -61,7 +62,7 @@ export function isAdhocColumn(column?: any): column is AdhocColumn {
     typeof column !== 'string' &&
     column?.sqlExpression !== undefined &&
     column?.label !== undefined &&
-    column?.expressionType === 'SQL'
+    (!isDefined(column?.expressionType) || column?.expressionType === 'SQL')
```

Comment:
Maybe the same logic? 

```suggestion
....
  return (
    typeof column !== 'string' &&
    column?.label !== undefined &&
    (column?.expressionType === null || column?.expressionType === 'SQL')
```

Reply:
I think we need to cover `undefined` here.  I'll change that to `column?.expressionType === undefined`